### PR TITLE
Allow user to choose a percentage of hosts to deploy to

### PR DESCRIPTION
**Pull request by spladug**
This is useful to say "go to 50% and wait" or something similar.

:eyeglasses: @kjoconnor @Deimos

### CODE REVIEW

**Comment by spladug** [comment]
example session:

```
ubuntu@rollingpin:~/src/rollingpin$ rollout  -h example 
*** starting deploy
[  example-00] connecting
[  example-00] success! all done
*** 1% done
*** waiting for input: e[x]it, [c]ontinue, [a]ll remaining, [p]ercentage
how far? (2-100) 50
... [snip] ...
*** 50% done
*** waiting for input: e[x]it, [c]ontinue, [a]ll remaining, [p]ercentage
...
```

File: rollingpin/frontends.py [hunk 1]
```diff
@@ -157,20 +165,25 @@ def __init__(self):
...
-        # go to single-character, no-echo mode
+        self.disable_echo()
+
+    def disable_echo(self):
         fileno = sys.stdin.fileno()
         self.old_termio_settings = termios.tcgetattr(fileno)
         tty.setcbreak(fileno)
 
+    def restore_terminal_settings(self):
+        # restore the terminal
```
Comment (Deimos) [inline rollingpin/frontends.py]: Tiny nit: can probably just drop this comment too, similar to what you did with `disable_echo()`.

File: rollingpin/frontends.py [hunk 2]
```diff
@@ -200,45 +226,52 @@ def on_sleep(self, host, count):
...
+                self.pause_after = 0
+                break
+            elif c == "x":
+                raise AbortDeploy("x pressed")
+            elif c == "c":
+                self.pause_after = 1
+                break
+            elif c == "p":
+                min_percent = self.percent_complete() + 1
+                prompt_input = yield self.console_input.raw_input("how far? (%d-100) " % min_percent)
```
Comment (Deimos) [inline rollingpin/frontends.py]: Nit: pretty long line, maybe two?

                prompt = "how far? (%d-100) " % min_percent
                prompt_input = yield self.console_input.raw_input(prompt)

**Comment by Deimos** [comment]
Looks great, just a couple of really tiny nits. :fish:

File: rollingpin/frontends.py (hunk 2 above)
Comment (spladug) [inline rollingpin/frontends.py]: gp, done

File: rollingpin/frontends.py (hunk 1 above)
Comment (spladug) [inline rollingpin/frontends.py]: kk

**Comment by kjoconnor** [comment]
Chiming in posthumously to say I love this, thanks!

**Pull request merged**